functests/ptp/ptp.go

```diff
@@ -108,4 +96,437 @@ var _ = Describe("ptp", func() {
 			}
 		})
 	})
+	var _ = Describe("PTP e2e tests", func() {
+		var runningPod *v1core.PodList
+		var ptpRunnungPods []v1core.Pod
+		var _ = Context("PTP Interfaces discovery", func() {
+
+			BeforeEach(func() {
+				var err error
+				ptpRunnungPods = nil
+				runningPod = nil
+				runningPod, err = client.Client.Pods(ptpLinuxDaemonNamespace).List(metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(len(runningPod.Items)).To(BeNumerically(">", 0), fmt.Sprint("linuxptp-daemon is not deployed on cluster"))
+				for _, pod := range runningPod.Items {
+					if podRole(pod, ptpSlaveNodeLabel) || podRole(pod, ptpGrandmasterNodeLabel) {
+						waitUntilLogIsDetected(pod, 180, "PTP capable NICs")
+						ptpRunnungPods = append(ptpRunnungPods, pod)
+					}
+				}
+				Expect(len(ptpRunnungPods)).To(BeNumerically(">", 0), fmt.Sprint("Fail to detect PTP slave/master pods on Cluster"))
+			})
+
+			// 25729
+			It("The interfaces support ptp can be discovered correctly", func() {
+
+				for _, pod := range ptpRunnungPods {
+					ptpSupportedInt := getPtpMasterSlaveAttachedInterfaces(pod)
+					Expect(len(ptpSupportedInt)).To(BeNumerically(">", 0), fmt.Sprint("Fail to detect PTP Supported interfaces on slave/master pods"))
+					ptpDiscoveredInterfaces := ptpDisoveredInterfaceList(nodePtpDeviceAPIPath + pod.Spec.NodeName)
+					matchCounter := len(ptpDiscoveredInterfaces)
+					Expect(matchCounter).To(BeNumerically("==", len(ptpSupportedInt)), fmt.Sprint("The interfaces discovered incorrectly"))
+					for _, inter := range ptpSupportedInt {
+						if Expect(ptpDiscoveredInterfaces).To(ContainElement(inter)) {
```

Comment:
This means that all the interfaces in `ptpSupportedInt` must be in `ptpDiscoveredInterfaces`, as we want their lenght to be equal and we want all the elements to be in the other list.
Can we get rid of matchCounter and fail as soon as one `inter` is not in the second list? I think you are doing something similar below in 25730
Also, can `inter` be something like `interface`?